hypha/apply/projects/views/payment.py

```diff
@@ -416,12 +419,35 @@ def dispatch(self, *args, **kwargs):
         return super().dispatch(*args, **kwargs)
 
     def get_media(self, *args, **kwargs):
-        file_pk = kwargs.get("file_pk")
-        if not file_pk:
-            return self.invoice.document
+        # check if the request is for a supporting document
+        if file_pk := kwargs.get("file_pk"):
+            document = get_object_or_404(self.invoice.supporting_documents, pk=file_pk)
+            return document.document
 
-        document = get_object_or_404(self.invoice.supporting_documents, pk=file_pk)
-        return document.document
+        # if not, then it's for invoice document
+        if (
+            self.invoice.status == APPROVED_BY_STAFF
+            and self.invoice.document.file.name.endswith(".pdf")
+        ):
+            if activities := Activity.actions.filter(
+                related_content_type__model="invoice",
+                related_object_id=self.invoice.id,
+                message__icontains="Approved by",
+            ).visible_to(self.request.user):
```

Comment:
It will pick all the Approved by ... statuses. If an invoice has been back and forth between finance and staff then it might show multiple `Approved by Staff` but I think only latest one should be fine. We might need distinct activities.  Also, we don't want to add this for invoices with status `Approved by finance`?
![image](https://github.com/HyphaApp/hypha/assets/23638629/d091a993-91c3-4169-871d-eccf5573b036)